test/e2e/network/network_policy.go

```diff
@@ -352,12 +486,552 @@ var _ = SIGDescribe("NetworkPolicy", func() {
+			var nsBserviceA, nsBserviceB *v1.Service
+			var nsBpodServerA, nsBpodServerB *v1.Pod
+
+			nsA := f.Namespace
+			nsBName := f.BaseName + "-b"
+			// The CreateNamespace helper uses the input name as a Name Generator, so the namespace itself
+			// will have a different name than what we are setting as the value of ns-name.
+			// This is fine as long as we don't try to match the label as nsB.Name in our policy.
+			nsB, err := f.CreateNamespace(nsBName, map[string]string{
+				"ns-name": nsBName,
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			// Creating pods and services in namespace-b
+			nsBpodServerA, nsBserviceA = createServerPodAndService(f, nsB, "ns-b-server-a", []int{80})
+			nsBpodServerB, nsBserviceB = createServerPodAndService(f, nsB, "ns-b-server-b", []int{80})
+
+			// Wait for Server with Service in NS-A to be ready
+			framework.Logf("Waiting for servers to come up.")
+			err = framework.WaitForPodRunningInNamespace(f.ClientSet, podServer)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Wait for Servers with Services in NS-B to be ready
+			err = framework.WaitForPodRunningInNamespace(f.ClientSet, nsBpodServerA)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = framework.WaitForPodRunningInNamespace(f.ClientSet, nsBpodServerB)
+			Expect(err).NotTo(HaveOccurred())
+
+			defer cleanupServerPodAndService(f, nsBpodServerA, nsBserviceA)
+			defer cleanupServerPodAndService(f, nsBpodServerB, nsBserviceB)
+
+			By("Creating a network policy for the server which allows traffic only to a server in different namespace.")
+			policy := &networkingv1.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: nsA.Name,
+					Name:      "allow-to-ns-b-server-a-via-namespace-selector",
+				},
+				Spec: networkingv1.NetworkPolicySpec{
+					// Apply this policy to the client
+					PodSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"pod-base-name": "client-a",
+						},
+					},
+					PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeEgress},
+					// Allow traffic only to server-a in namespace-b
+					Egress: []networkingv1.NetworkPolicyEgressRule{{
+						To: []networkingv1.NetworkPolicyPeer{
+							{
+								NamespaceSelector: &metav1.LabelSelector{
+									MatchLabels: map[string]string{
+										"ns-name": nsBName,
+									},
+								},
+								PodSelector: &metav1.LabelSelector{
+									MatchLabels: map[string]string{
+										"pod-base-name": nsBpodServerA.ObjectMeta.Labels["pod-base-name"],
+									},
+								},
+							},
+							// Need to allow traffic to DNS for name relsolution
+							{
+								NamespaceSelector: &metav1.LabelSelector{
+									MatchLabels: map[string]string{},
+								},
+								PodSelector: &metav1.LabelSelector{
+									MatchLabels: map[string]string{
+										"k8s-app": "kube-dns",
```

Comment:
The OpenShift e2e tests pull in a bunch of the upstream kubernetes e2e tests, but are run in a slightly different environment where `podSelector: { matchLabels: { k8s-app: kube-dns } }` would not be correct. I don't know if anyone else does anything similar...

But anyway, there's an easy workaround for that (which is also more future-proof just within kubernetes): just allow egress to all destinations on port 53.

Reply:
Acknowledged and done.